consensus/src/vetomint.rs

```diff
@@ -0,0 +1,105 @@
+mod progress;
+#[cfg(test)]
+mod tests;
+
+/// An index of the validator, which is for a single height. (Mapping from the actual public key to the index may differ for different heights.)
+pub type ValidatorIndex = usize;
+/// An identifier of the block, which is uniquely mapped to a block. Like `ValidatorIndex`, it is for a single height. (Mapping from the actual block to the index may differ for different heights.)
+pub type BlockIdentifier = usize;
+/// A UNIX timestamp measured in milliseconds.
+pub type Timestamp = i64;
+
+/// An event that (potentially) triggers a state transition of `StateMachine`.
+///
+/// Note that there is no cryptography-related info here, because it's
+/// the lower layer's responsibility to verifiy and refine the raw messages (containing such cryptography-related info) into this abstracted data.
+/// Also all the identifiers (for blocks and validators) become integer indices here, and
+/// the lower layer will keep the mapping from the actual data to the indices.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ConsensusEvent {
+    BlockProposal {
+        proposal: BlockIdentifier,
+        proposer: ValidatorIndex,
+        round: usize,
+        /// Whether the operator of this consensus is in favor of this proposal.
+        favor: bool,
+    },
+    Prevote {
+        proposal: BlockIdentifier,
```

Comment:
Given a fixed round, proposal is fixed as well. Do we need this field?

Reply:
The proposal's not fixed, as the earlier comment answered.